Allow the Homepage to be published

36degrees opened the pull request:
This updates the topic presenter to include a link to the homepage as a `parent` if the topic is set to be included on the homepage (the communities topic won't be, and we need a way to filter that out)

We don't currently have a good way to publish the homepage, so the HomepagePresenter only exists to define a content id for the homepage. I'm not sure of the implications of creating links to things that don't exist yet, so that's why this PR is marked as do not merge.

thehenster commented on spec/controllers/topics_controller_spec.rb:
```diff
@@ -0,0 +1,31 @@
+      end
+    end
+  end
+end
```
Why did you opt for writing controller tests rather than add to the feature spec?

36degrees replied:
Erm.

Honestly, I have no idea. It probably seemed like a good idea at the time, but I wrote this a month ago. These can probably be removed, if they're not adding anything.

thehenster commented on spec/features/topic_spec.rb:
```diff
@@ -11,6 +11,12 @@
+    visit new_topic_path
+
+    expect(page).to have_checked_field 'Include on homepage?'
+  end
```
:+1:

thehenster commented on spec/presenters/homepage_presenter_spec.rb:
```diff
@@ -0,0 +1,56 @@
+  it 'includes the rendering and publishing apps' do
+    payload = described_class.new.content_payload
+
+    expect(payload).to include \
```
The `\` here is kinda neat. Let's give it a whirl for a bit. Brackets are the more normal option.

36degrees replied:
I like normal. Not sure why I decided to do this instead. Happy to bracket it up…

thehenster commented on app/presenters/topic_presenter.rb:
```diff
@@ -36,6 +36,7 @@ def links_payload
         linked_items: topic.guides.map(&:content_id),
         content_owners: content_owner_content_ids,
         organisations: [ServiceManualPublisher::GDS_ORGANISATION_CONTENT_ID],
+        parent: parents
```
I still think doing this is our best option rather than tackling a custom link expansion.

We should probably add to the schema example: https://github.com/alphagov/govuk-content-schemas/blob/master/formats/service_manual_topic/frontend/examples/service_manual_topic.json

36degrees replied:
[This is done](https://github.com/alphagov/govuk-content-schemas/pull/420).

thehenster commented on lib/tasks/migrate_homepage.rake:
```diff
@@ -0,0 +1,38 @@
+    "Publishing the homepage will make the old service manual inaccessible. Continue?",
+    :red
+  )
+  exit unless cli.agree "Are you sure you wish to continue?"
```
Faaancy :)

thehenster commented on lib/tasks/migrate_homepage.rake:
```diff
@@ -0,0 +1,38 @@
+  )
+  exit unless cli.agree "Are you sure you wish to continue?"
+
+  publishing_api_v1 = GdsApi::PublishingApi.new(
```
We're using v1 because of something to do with the path reservation yes? Could you remind me quickly.. I has forgot.

36degrees replied:
[Yep](https://github.com/alphagov/publishing-api/blob/master/doc/api.md#put-pathsbase_path).

36degrees commented:
@mcmillan would you mind taking another look at this? I've rebased it, but the last two commits are entirely new:

- Changed the route type for the homepage from exact to prefix, so any requests for anything starting with '/service-manual' come to our app. We'll therefore be able to e.g. see 404s in our logs.
- Realised we need to republish topics as part of migration so that the relationship with the homepage exists.
